client/src/main/java/io/pravega/client/stream/impl/CheckpointState.java

```diff
@@ -97,12 +97,12 @@ void beginNewCheckpoint(String checkpointId, Set<String> currentReaders, Map<Seg
     }
     
     String getCheckpointForReader(String readerName) {
-        OptionalInt min = getCheckpointsForReader(readerName).stream().mapToInt(checkpoints::indexOf).min();
-        if (min.isPresent()) {
-            return checkpoints.get(min.getAsInt());
-        } else {
+        List<String> checkpointsForReader = getCheckpointsForReader(readerName);
+        if (checkpointsForReader.isEmpty()) {
             return null;
         }
+        OptionalInt min = checkpointsForReader.stream().mapToInt(checkpoints::indexOf).min();
```

Comment:
Replace this with a standard for loop. It will be faster.

Reply:
Fixed.